crates/cext/src/circuit.rs

```diff
@@ -605,3 +605,65 @@ pub unsafe extern "C" fn qk_circuit_to_python(circuit: *mut CircuitData) -> *mut
             .into_ptr()
     }
 }
+
+/// @ingroup QkCircuit
+///
+/// C compatible enum for DelayUnit
+#[repr(C)]
+pub enum QkDelayUnit {
+    S,
+    MS,
+    US,
+    NS,
+    PS,
+    DT,
+}
+
+/// @ingroup QkCircuit
+/// Append a delay instruction to the circuit.
+///
+/// @param circuit A pointer to the circuit to add the delay to.
+/// @param qubit The ``uint32_t`` index of the qubit to apply the delay to.
+/// @param duration The duration of the delay.
+/// @param unit A C enum representing the unit of the duration.
+///
+/// # Example
+///
+///     QkCircuit *qc = qk_circuit_new(1, 0);
+///     qk_circuit_delay(qc, 0, 100.0, QkDelayUnit_NS);
+///     
+/// # Safety
+///
+/// Behavior is undefined if ``circuit`` is not a valid, non-null pointer to a ``QkCircuit``.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_circuit_delay(
+    circuit: *mut CircuitData,
+    qubit: u32,
+    duration: f64,
+    unit: QkDelayUnit,
+) -> ExitCode {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let circuit = unsafe { mut_ptr_as_ref(circuit) };
+
+    let delay_unit_variant = match unit {
+        QkDelayUnit::S => DelayUnit::S,
```

Comment:
We typically don't add the `Qk` prefix to the structs manually (see e.g. `SparseTerm` or `OpCounts` in the `cext` crate), and have `cbindgen` add this prefix. It would be more consistent to do the same here, but I'm not super set on this 🙂